include/zephyr/bluetooth/mesh/access.h

```diff
@@ -904,18 +916,16 @@ struct bt_mesh_model {
 	/** Model callback structure. */
 	const struct bt_mesh_model_cb * const cb;
 
-#ifdef CONFIG_BT_MESH_MODEL_EXTENSIONS
-	/* Pointer to the next model in a model extension list. */
-	struct bt_mesh_model *next;
-#endif
-
 #ifdef CONFIG_BT_MESH_LARGE_COMP_DATA_SRV
 	/* Pointer to the array of model metadata entries. */
-	struct bt_mesh_models_metadata_entry **metadata;
+	struct bt_mesh_models_metadata_entry ** const metadata;
 #endif
 
+	/** Model ctx variables. */
+	struct bt_mesh_model_ctx * const ctx;
+
 	/** Model-specific user data */
-	void *user_data;
+	void * const user_data;
```

Comment:
Otherwise this won't allow to declare `bt_mesh_model` as constant. The only limitation from what I see is that customers won't be able to set `user_data` at runtime, but not sure if this is needed. In any case it can be bypassed I guess.

Reply:
If structure is linked into flash memory due to `const` modifier, how is it possible to bypass?

Reply:
Probably move `user_data` into context structure?

Reply:
Like this:

Reply:
Should it be added here as recommendation in field description?

Reply:
We can, but I'm not sure it is necessary. I don't have strong argument against, but I think that fewer users or no one do this (changes user data at runtime).

Reply:
I guess it's worth. Since it is a public API change, it breaks runtime user data changing. We need to suggest alternative approach for customers.